examples/custom-server/src/server.ts

```diff
@@ -0,0 +1,59 @@
+import dotenv from 'dotenv'
+import next from 'next'
+import nextBuild from 'next/dist/build'
+import path from 'path'
+
+dotenv.config({
+  path: path.resolve(__dirname, '../.env'),
+})
+
+import express from 'express'
+import payload from 'payload'
+
+import { seed } from './seed'
+
+const app = express()
+const PORT = process.env.PORT || 3000
+
+const start = async (): Promise<void> => {
+  await payload.init({
+    secret: process.env.PAYLOAD_SECRET || '',
+    mongoURL: process.env.MONGODB_URI || '',
+    express: app,
+    onInit: () => {
+      payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`)
+    },
+  })
+
+  if (process.env.PAYLOAD_SEED === 'true') {
+    payload.logger.info('---- SEEDING DATABASE ----')
+    await seed(payload)
+  }
+
+  if (process.env.NEXT_BUILD) {
+    app.listen(PORT, async () => {
+      payload.logger.info(`Next.js is now building...`)
+      // @ts-expect-error
+      await nextBuild(path.join(__dirname, '../'))
+      process.exit()
+    })
+  }
```

Comment:
Is this code still relevant? When building using `next build` this file is not run.
```suggestion
```

Reply:
Good catch! I just pushed https://github.com/payloadcms/payload/pull/2920/commits/eaf1e310b7c019969e1a9fbc20c9043ba21e5c1d which _does_ execute this file. This is necessary because Payload is required to be running in order for the front-end to have endpoints to hit.